core/arch/arm/kernel/kern.ld.S

```diff
@@ -257,6 +256,17 @@ SECTIONS
 	__pageable_start = __text_init_start;
 	__pageable_end = __pageable_part_end;
 
+	/*
+	 * Assign a safe spot to store the hashes of the pages before
+	 * heap is initialized.
+	 */
+	__tmp_hashes_start = __rodata_init_end;
+	__tmp_hashes_size = (((__pageable_end - __pageable_start - 1) /
+				(4 * 1024)) + 1) * 0x20;
```

Comment:
To have the correct size, __pageable_end and __pageable_start must be 4KB aligned, which is true. In this case, I guess that the following is sufficient:
`__tmp_hashes_size = (((__pageable_end - __pageable_start ) / (4 * 1024))) * 0x20;`